advisor-go/src/quixo/simulation/simulation.go

```diff
@@ -16,35 +16,54 @@ func GetBestMoveForPlayer(board game.Board) Move {
 	return getBestMove(board, false)
 }
 
-// GetWorstMoveForPlayer return the move that will benefit the most to the opponent player
+// GetWorstMoveForPlayer return the move that will benefit the most for the opponent player
 func GetWorstMoveForPlayer(board game.Board) Move {
 	return getBestMove(board, true)
 }
 
+// GetMoveThatMakeMyOpponentLose return the move that will be the worst for the opponent
+func GetMoveThatMakeMyOpponentLose(board game.Board) Move {
+	isSmallerThan := func(int1 int, int2 int) bool { return int1 < int2 }
+	if hasCubeSelected(board) {
+		return getMoveWithCubeSelected(board, true, isSmallerThan)
+	}
+	return getMoveWithNoCubeSelected(board, true, isSmallerThan)
+}
+
 func getBestMove(board game.Board, forOpponentPlayer bool) Move {
+	isGreaterThan := func(int1 int, int2 int) bool { return int1 > int2 }
+
 	if hasCubeSelected(board) {
-		return getBestMoveWithCubeSelected(board, forOpponentPlayer)
+		return getMoveWithCubeSelected(board, forOpponentPlayer, isGreaterThan)
 	}
-	return getBestMoveWithNoCubeSelected(board, forOpponentPlayer)
+	return getMoveWithNoCubeSelected(board, forOpponentPlayer, isGreaterThan)
 }
 
-func getBestMoveWithCubeSelected(board game.Board, forOpponentPlayer bool) Move {
-	destination, _ := getBestDestinationWithScore(board, board.SelectedCube, forOpponentPlayer)
+func getMoveWithCubeSelected(
+	board game.Board,
+	forOpponentPlayer bool,
+	comparator func(int, int) bool,
```

Comment:
to compare what?

Reply:
It's the score comparator. I'll change that to make it clearer